sympy/diffgeom/diffgeom.py

```diff
@@ -1131,10 +1131,8 @@ def _contravariant_slots(self,T):
                 slots.append(k)
         return slots
 
-    def __init__(self, **kwargs):
-        self.CoordSystem = kwargs['coordinate_system']
-        variance = kwargs['variance']
-        component_array=kwargs['components']
+    def __init__(self, components, variance, coordinate_system):
```

Comment:
This looks good. There is more information on coding and documentation conventions in SymPy in the wiki articles on [development workflow](https://github.com/sympy/sympy/wiki/Development-workflow#id22) and [documentation](https://github.com/sympy/sympy/wiki/Writing-Documentation). In particular, documentation of classes and methods follows the [Numpy/Scipy guide](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard).

Reply:
What about renaming `variance ==> valence`. Valence refers to both covariance and contravariance.